src/main/scala/com/fasterxml/jackson/module/scala/ScalaObjectMapper.scala

```diff
@@ -0,0 +1,372 @@
+    readerFor(constructType[T])
+  }
+
+  /**
+   * Factory method for constructing [[com.fasterxml.jackson.databind.ObjectReader]] that will
+   * deserialize objects using specified JSON View (filter).
+   */
+  def readerWithView[T: Manifest]: ObjectReader = {
+    readerWithView(manifest[T].runtimeClass)
+  }
+
+  /*
+   **********************************************************
+   * Extended Public API: convenience type conversion
+   **********************************************************
+   */
+
+  /**
+   * Convenience method for doing two-step conversion from given value, into
+   * instance of given value type. This is functionality equivalent to first
+   * serializing given value into JSON, then binding JSON data into value
+   * of given type, but may be executed without fully serializing into
+   * JSON. Same converters (serializers, deserializers) will be used as for
+   * data binding, meaning same object mapper configuration works.
+   *
+   * @throws IllegalArgumentException If conversion fails due to incompatible type;
+   *                                  if so, root cause will contain underlying checked exception data binding
+   *                                  functionality threw
+   */
+  def convertValue[T: Manifest](fromValue: Any): T = {
+    convertValue(fromValue, constructType[T])
+  }
+
+  /*
+   **********************************************************
+   * Extended Public API: JSON Schema generation
+   **********************************************************
+   */
+
+  /**
+   * Generate <a href="http://json-schema.org/">Json-schema</a>
+   * instance for specified class.
+   *
+   * @tparam T The class to generate schema for
+   * @return Constructed JSON schema.
+   */
+  @deprecated("JsonSchema is deprecated in favor of JsonFormatVisitor", "2.2.2")
+  def generateJsonSchema[T: Manifest]: JsonSchema = {
+    generateJsonSchema(manifest[T].runtimeClass)
+  }
+
+  /**
+   * Method for visiting type hierarchy for given type, using specified visitor.
+   * <p>
+   * This method can be used for things like
+   * generating <a href="http://json-schema.org/">Json Schema</a>
+   * instance for specified type.
+   *
+   * @tparam T Type to generate schema for (possibly with generic signature)
+   *
+   * @since 2.1
+   */
+  def acceptJsonFormatVisitor[T: Manifest](visitor: JsonFormatVisitorWrapper): Unit = {
+    acceptJsonFormatVisitor(manifest[T].runtimeClass, visitor)
+  }
+
+  private def isArray(c: Class[_]): Boolean = {
+    c.isArray
+  }
+
+  private val MAP = classOf[collection.Map[_,_]]
+  private def isMapLike(c: Class[_]): Boolean = {
+    MAP.isAssignableFrom(c)
+  }
+
+  private val OPTION = classOf[Option[_]]
+  private def isReference(c: Class[_]): Boolean = {
+    OPTION.isAssignableFrom(c)
+  }
+
+  private val ITERABLE = classOf[collection.Iterable[_]]
+  private def isCollectionLike(c: Class[_]): Boolean = {
+    ITERABLE.isAssignableFrom(c)
+  }
+}
```

Comment:
nit: Could you add a newline here?
```suggestion
}

```